File: 118-pascal-triangle.ts
// numRows is the argument LeetCode uses
let numRows: number = 5;

let result: number[][] = [];

for (let row = 1; row < numRows + 1; row++) {
  let temp: number[] = [];
  for (let column = 0; column < row; column++) {
    if (row <= 2) {
      temp.push(1);
    }
    else{
      if ((column - 1 < 0) || (column > row - 2)){
        temp.push(1)
      }
      else{
        temp.push(result[row - 2][column - 1] + result[row - 2][column])
      }
    }
  }
  result.push(temp)
}

console.log(result)
// return result

File: 118-pascal-triangle.py
# numRows is the argument LeetCode uses
numRows = 5

result = []

for row in range(1, numRows + 1, 1):
    temp = []
    for column in range(row):
        if row <= 2:
            temp.append(1)
        else:
            if column - 1 < 0 or column > row - 2:
                temp.append(1)
            else:
                temp.append(result[row - 2][column - 1] + result[row - 2][column])
    result.append(temp)

print(result)
# return result
